api/client/build.go

```diff
@@ -131,10 +130,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		return fmt.Errorf("cannot canonicalize dockerfile path %s: %v", relDockerfile, err)
 	}
 
-	var includes = []string{"."}
+	var (
+		includes = []string{"."}
+		excludes []string
+	)
 
-	excludes, err := utils.ReadDockerIgnore(path.Join(contextDir, ".dockerignore"))
-	if err != nil {
+	f, err := os.Open(filepath.Join(contextDir, ".dockerignore"))
```

Comment:
interesting. Can you elaborate more on the future use of the Context interface? Are you planning on supporting a build context that isn't actually on disk?  Just trying to understand when we wouldn't have a real file to point to.